### DISCUSSION TITLE
Problema con GetIssuedDocumentPreCreateInfo

### DISCUSSION THREAD

**Discussion by e-Labora** [fattureincloud/api#190] [Q&A] [answered]
Buongiorno.
Ho il seguente problema: vorrei utilizzare GetIssuedDocumentPreCreateInfo, ma il codice seguente (VB.NET) non mi restituisce nulla.
Controllando i log dal nostro account FattureInCloud vedo tutte risposte 200, ma quando cerco di leggere la risposta ottengo l'errore "Object Reference not set to an instance of an object".

```
Dim strFattureInCloud_Intestato As String = objPTWebApp.Get_Parametro_Valore("BANCA_INTESTATO")

Dim objFICConfig As Configuration = New Configuration()
objFICConfig.BasePath = "https://api-v2.fattureincloud.it/"
objFICConfig.AccessToken = "il_mio_access_token"

Dim objIssuedDocumentsAPIInstance = New IssuedDocumentsApi(objFICConfig)

Dim objFICPreCreateInfo_Res As GetIssuedDocumentPreCreateInfoResponse

objFICPreCreateInfo_Res = objIssuedDocumentsAPIInstance.GetIssuedDocumentPreCreateInfo("il_mio_username", "invoice")
```

In questi giorni ho già usato altre chiamate come ad esempio List Clients, Modify Client, Create Client senza problemi.

**Comment by emajo** [comment, marked as answer]
Ciao @e-Labora il problema potrebbe essere legato a numerations e dn_numerations, in questa [discussion](https://github.com/fattureincloud/api/discussions/163) è spiegato in maniera più dettagliata la problematica e dei workaround temporanei in quanto non sappiamo ancora quando verrà rilasciato il fix.

Facci sapere se ti risulta che il problema sia questo o, in caso non lo fosse, ti consiglio di contattare il nostro supporto clienti per andare a vedere più in dettaglio la problematica.
Grazie,
Emanuele.

**Comment by e-Labora** [comment]
Ciao. 
Vi confermo che il problema è proprio quello evidenziato.
Creando una fattura ed una DDT fittizie la risposta non è più un oggetto null.
Ne approfitto per chiedervi due cose:

1) Come faccio per recuperare le coppie chiave valore (numerazione/ultimo numero fattura)che ho evidenziato nell'immagine tramite l'SDK? 

2) Se cerco di recuperare l'id del template di default (evidenziato nell'immagine) tramite

`Dim strTemplateID As String = objFICPreCreateInfo_Res.Data.DefaultValues.DefaultTemplate.Id`

mi restituisce l'eccezione "Object Reference not set to an instance of an object", mentre se ad esempio uso

`Dim strTemplateID As String = objFICPreCreateInfo_Res.Data.DefaultValues.DnTemplate.Id`

la variabile viene valorizzata correttamente.

![fattureincloud](https://user-images.githubusercontent.com/10773621/207372697-2d49332f-121e-4b01-986d-fcc58b10447e.jpg)

**Reply by emajo** [reply]
Per quanto riguarda il primo punto sarebbe da iterare le chiavi del Dictionary che viene restituito, solitamente non è necessaria un operazione del genere in quanto in creazione di documento se non specificato il number viene assegnato automaticamente (numero incrementale), se è necessario recuperare tali valori per altri scopi posso abbozzare un esempio.
Per il secondo punto invece credo basti fare `objFICPreCreateInfo_Res.Data.DefaultValues.Template.Id`

**Reply by e-Labora** [reply]
Chiedo scusa, non avevo letto bene la documentazione e credevo che number e numeration andassero valorizzati obbligatoriamente in fase di creazione di un documento.

Per il secondo punto ci avevo già provato, ma l'intellisense mi segnala che 'Template' non è un membro di IssuedDocumentPreCreateInfoDefaultValue e mi propone solo "AiTemplate", "DnTemplate" e "DefaultTemplate"
